src/Resolver/Asset/ResourceSymbolicImageHierachyResolver.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\GraphQL\Search\Resolver\Asset;
+
+use Atoolo\GraphQL\Search\Resolver\UrlRewriter;
+use Atoolo\GraphQL\Search\Resolver\UrlRewriterType;
+use Atoolo\GraphQL\Search\Types\Asset;
+use Atoolo\GraphQL\Search\Types\SymbolicImage;
+use Atoolo\Resource\Resource;
+use Atoolo\Resource\ResourceHierarchyLoader;
+use Atoolo\Resource\ResourceHierarchyWalker;
+use Overblog\GraphQLBundle\Definition\ArgumentInterface;
+
+class ResourceSymbolicImageHierachyResolver implements ResourceAssetResolver, ResourceSymbolicImageResolver
+{
+    public function __construct(
+        private readonly UrlRewriter $urlRewriter,
+        private readonly ResourceHierarchyLoader $hierarchyLoader
+    ) {
+    }
+
+    public function getAsset(
+        Resource $resource,
+        ArgumentInterface $args
+    ): ?Asset {
+        return $this->getSymbolicImage($resource, $args);
+    }
```

Comment:
the `ResourceSymbolicImageHierachyResolver` does not need a `getAsset()` method.

Reply:
It does, because it implements the `ResourceAssetResolver` interface. That way, it can be used by the `ResourceAssetResolverChain`. So this is intended as well.

Note that I added two interfaces: `ResourceAssetResolver` and `ResourceSymbolicImageResolver`. The `ResourceSymbolicImageHierachyResolver` implements both.

Reply:
Aber die `ResourceAssetResolverChain`brauch das nicht. Sie bekommt den ResourceSymbolicImageResolver über das Tagging wie alle anderen AssetResolver auch. Das finde auch auch richtig so:

```yml
    Atoolo\GraphQL\Search\Resolver\Asset\ResourceSymbolicImageHierachyResolver:
      arguments:
        - '@atoolo_graphql_search.resolver.url_rewriter'
        - '@atoolo_resource.navigation_hierarchy_loader'
      tags:
        - { name: 'atoolo_graphql_search.resolver.asset', priority: 10 }
```